### PR TITLE
Add PMD Parser

### PR DESCRIPTION
**Pull request by jis0324**
Added PMD parser.

### CODE REVIEW

File: dojo/unittests/tools/test_pmd_parser.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        testfile = open("dojo/unittests/scans/pmd/pmd_one_vuln.csv")
+        parser = PmdParser()
+        findings = parser.get_findings(testfile, Test())
+        self.assertEqual(1, len(findings))
+
+    def test_parse_file_with_multiple_vuln_has_multiple_finding(self):
+        testfile = open("dojo/unittests/scans/pmd/pmd_many_vulns.csv")
+        parser = PmdParser()
+        findings = parser.get_findings(testfile, Test())
+        self.assertEqual(9, len(findings))
```
Comment (damiencarol) [inline dojo/unittests/tools/test_pmd_parser.py:24]: Please add more check on the attributes of some findings.
Like for ex test on file_path / line attributes

File: dojo/tools/pmd/parser.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    def get_findings(self, filename, test):
+        dupes = dict()
+
+        content = filename.read()
+        if type(content) is bytes:
+            content = content.decode('utf-8')
+        reader = csv.DictReader(io.StringIO(content), delimiter=',', quotechar='"')
+        csvarray = []
+
+        for row in reader:
```
Comment (damiencarol) [inline dojo/tools/pmd/parser.py:27]: Could you get ride of the double loop?
I know some parser do it but it's not a good thing :/ 
I'm sure you can do something better :)

**Comment by damiencarol** [comment]
@jis0324 seems your unit tests is not good:
```
uwsgi_1         | ======================================================================
uwsgi_1         | FAIL: test_parse_file_with_multiple_vuln_has_multiple_finding (dojo.unittests.tools.test_pmd_parser.TestPMDParser)
uwsgi_1         | ----------------------------------------------------------------------
uwsgi_1         | Traceback (most recent call last):
uwsgi_1         |   File "/app/dojo/unittests/tools/test_pmd_parser.py", line 24, in test_parse_file_with_multiple_vuln_has_multiple_finding
uwsgi_1         |     self.assertEqual(9, len(findings))
uwsgi_1         | AssertionError: 9 != 15
```

**Comment by jis0324** [comment]
Hi, @damiencarol 
It seems to be passed all checkers.

File: dojo/unittests/tools/test_pmd_parser.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        testfile = open("dojo/unittests/scans/pmd/pmd_one_vuln.csv")
+        parser = PmdParser()
+        findings = parser.get_findings(testfile, Test())
+        self.assertEqual(1, len(findings))
+
+    def test_parse_file_with_multiple_vuln_has_multiple_finding(self):
+        testfile = open("dojo/unittests/scans/pmd/pmd_many_vulns.csv")
+        parser = PmdParser()
+        findings = parser.get_findings(testfile, Test())
+        self.assertEqual(15, len(findings))
```
Comment (damiencarol) [inline dojo/unittests/tools/test_pmd_parser.py:24]: Could you add more check on attributes of the result?
Like:
```
self.assertEqual("<foo>", findings[0].severity)
```

**Review by damiencarol — APPROVED** [review]
LGTM. If you can add more unit tests it will be perfect.

File: dojo/unittests/tools/test_pmd_parser.py (hunk 3 above)
Comment (jis0324) [reply dojo/unittests/tools/test_pmd_parser.py:24]: I think i am not familiar with that, so I feel i am not confident.

Comment (damiencarol) [reply dojo/unittests/tools/test_pmd_parser.py:24]: it's ok. no pb. will try to add it today.

Comment (jis0324) [reply dojo/unittests/tools/test_pmd_parser.py:24]: Great. Thanks.
To pass a PR, it is too difficult. :(

Comment (damiencarol) [reply dojo/unittests/tools/test_pmd_parser.py:24]: no, you did a great job !

Comment (damiencarol) [reply dojo/unittests/tools/test_pmd_parser.py]: added some checks

Comment (jis0324) [reply dojo/unittests/tools/test_pmd_parser.py]: Yeah, I checked.
Great.
Thanks for your help. 👍

**Review by valentijnscholten — APPROVED** [review]

**Pull request merged**